Sources/SwiftProtobuf/AsyncMessageSequence.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Sources/SwiftProtobuf/AsyncMessageSequence.swift - Async sequence over binary delimited protobuf
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// An async sequence of messages decoded from a binary delimited protobuf stream.
+///
+// -----------------------------------------------------------------------------
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+public struct AsyncMessageSequence<Base: AsyncSequence, M: Message>:
+  AsyncSequence, AsyncIteratorProtocol where Base.Element == UInt8 {
```

Comment:
Please make the iterator a separate struct instead of conforming the sequence to iterator as well

Reply:
Done